### PR TITLE
LL-2217 Upgrade dependencies

### PR DESCRIPTION
**Pull request by LukeSugiura**
### References
- [React Navigation v5: upgrading-from-4.x](https://reactnavigation.org/docs/en/upgrading-from-4.x.html#configuring-the-navigator)

### TODO
- [x] upgrade deps except react-native or @ledgerhq related modules
- [x] create ScreenName, NavigationName const
- [x] create navigation components
- [x] import `ScrollView`, `FlatList`… from `react-native` instead of `react-navigation`
- [x] import `SafeAreaView` from `react-native-safe-area-context`
- [x] create `useNavigationInterceptor`
- [x] replace string screen/navigator name with obj from `src/const/navigation.js`
~- [ ] fix circular dependencies with GeneratedFamilyScreenName~
- [x] replace `static navigationOptions`
- [x] replace `getParams` with `route.params` -> ~make sure components using `navigation.getParam` are wrapped with `withNavigation` HOC~
- [x] replace `navigation.emit("refocus")` with `useScrollToTop` hook
  - find the way to pass ref to both ScrollView and header component
- [x] reimplement this logic: https://github.com/LedgerHQ/ledger-live-mobile/blob/develop/src/navigators.js#L283
- [x] delete old navigation logic files
  - src/navigation.js
  - src/screens/OnboardingOrNavigator.js
- [x] delete old navigation deps
  - react-navigation
  - react-navigation-stack
  - react-navigation-tabs
- [x] fix styling issues
- [x] get rid of navigation.state.params
- [x] scrollToTop on Portfolio ~and Accounts tab~
- [x] remove navigation.dismiss

### CODE REVIEW

**Comment by LukeSugiura** [comment]
### Styling issues
- [ ] header height
header-height
- [ ] header margin
header-margin
- [ ] side padding
side-padding
- [ ] side padding
side-padding1
- [ ] tab icon
tab-icon

**Comment by LukeSugiura** [comment]
*Current outdated dependencies*

Screen Shot 2020-02-21 at 4 21 53 PM

**Comment by LukeSugiura** [comment]
- [ ] 🐞After upgrade react-native, it gets build from xcode but not from react-native cli

File: package.json [hunk 1]
```diff
@@ -72,6 +72,7 @@
     "@segment/analytics-ios": "github:LedgerHQ/analytics-ios#efb4cd1771dab568422473fd680ffb748b102f07",
     "@segment/analytics-react-native": "^0.1.0-beta.0",
     "@tradle/react-native-http": "^2.0.0",
+    "add": "^2.0.6",
```
Comment (gre) [inline package.json]: @IAmMorrow  is that you :trollface:

**Comment by LukeSugiura** [comment]
### Bugs
- [ ] Settings -> General -> Password lock
- [ ] Settings -> Help -> Set up device
- [ ] Transfer -> Buy crypto assets
- [ ] Settings -> Debug -> Mock & Test
- [ ] No response: Portfolio/Accounts -> Add accounts/Import desktop accounts

### Header style bugs
- [ ] Password lock -> Set password

**Comment by LukeSugiura** [comment]
[ ] Android build failed
 
Screen Shot 2020-03-05 at 7 39 42 PM

**Comment by Arnaud97234** [comment]
> ### Bug found
> iOS - Password lock ON
> 
> When user open the app and submit the password form (Valid or incorrect), he is asked for the OS native password. Only then the LL password check is done.
> 
> _PS: On the screen record below the password input is blank but there is actually a password that is entered._
> 
> ### How to reproduce
> ezgif com-resize

**Comment by LukeSugiura** [comment]
### Ref
- https://github.com/oblador/react-native-keychain/issues/316#issuecomment-602581214

**Pull request merged**